app/test_main.py

```diff
@@ -1 +1,60 @@
-# write your code here
+import pytest
+from typing import Any
+from unittest import mock
+
+from app.main import can_access_google_page
+
+
+@pytest.fixture
+def mocked_validation() -> Any:
+    with mock.patch("app.main.valid_google_url") as mocked_validation:
+        yield mocked_validation
+
+
+@pytest.fixture
+def mocked_connection() -> Any:
+    with mock.patch("app.main.has_internet_connection") as mocked_connection:
+        yield mocked_connection
+
+
+def test_can_access_google_page_when_validation_and_connection(
+        mocked_validation: Any,
```

Comment:
You can write one test instead of four and parametrize inputs since your test do the same thing but with different params.
Also don't forget to test that your mocked functions has or has not been called